pkg/cache/shares.go

```diff
@@ -160,11 +165,42 @@ func UpdateSharedSecret(share *sharev1alpha1.SharedSecret) {
 	br := share.Spec.SecretRef
 	key := BuildKey(br.Namespace, br.Name)
 	secretsWithShare.Delete(key)
-	AddSharedSecret(share)
+	return
+}
+
+func NamespacesWithSharedConfigMaps() []string {
+	namespacesMap := map[string]bool{}
```

Comment:
```suggestion
	namespacesMap := map[string]struct{}
```

You can save some memory here by using an empty struct instead of a bool, then just do 
```
			namespacesMap[scm.Spec.ConfigMapRef.Namespace] = struct{}{}
```

Same thing with `NamespacesWithSharedSecrets` below.

Reply:
updated @coreydaley